code/game/objects/items/RCD.dm

```diff
@@ -1104,108 +1104,227 @@ GLOBAL_VAR_INIT(icon_holographic_window, init_holographic_window())
-	/obj/machinery/plumbing/pill_press = 20,
-	/obj/machinery/plumbing/acclimator = 10,
-	/obj/machinery/plumbing/bottler = 50,
-	/obj/machinery/plumbing/disposer = 10,
-	/obj/machinery/plumbing/fermenter = 30,
-	/obj/machinery/plumbing/filter = 5,
-	/obj/machinery/plumbing/grinder_chemical = 30,
-	/obj/machinery/plumbing/liquid_pump = 35,
-	/obj/machinery/plumbing/splitter = 5,
-	/obj/machinery/plumbing/sender = 20,
-	/obj/machinery/iv_drip/plumbing = 20
-)
+		// Note that the list MUST include fluid ducts.
+		/obj/machinery/duct = 1,
+		/obj/machinery/plumbing/input = 5,
+		/obj/machinery/plumbing/output = 5,
+		/obj/machinery/plumbing/tank = 20,
+		/obj/machinery/plumbing/synthesizer = 15,
+		/obj/machinery/plumbing/reaction_chamber = 15,
+		/obj/machinery/plumbing/buffer = 10,
+		//Above are the most common machinery which is shown on the first cycle. Keep new additions below THIS line, unless they're probably gonna be needed alot
+		/obj/machinery/plumbing/layer_manifold = 5,
+		/obj/machinery/plumbing/pill_press = 20,
+		/obj/machinery/plumbing/acclimator = 10,
+		/obj/machinery/plumbing/bottler = 50,
+		/obj/machinery/plumbing/disposer = 10,
+		/obj/machinery/plumbing/fermenter = 30,
+		/obj/machinery/plumbing/filter = 5,
+		/obj/machinery/plumbing/grinder_chemical = 30,
+		/obj/machinery/plumbing/liquid_pump = 35,
+		/obj/machinery/plumbing/splitter = 5,
+		/obj/machinery/plumbing/sender = 20,
+		/obj/machinery/iv_drip/plumbing = 20
+	)
 
 ///pretty much rcd_create, but named differently to make myself feel less bad for copypasting from a sibling-type
 /obj/item/construction/plumbing/proc/create_machine(atom/A, mob/user)
 	if(!machinery_data || !isopenturf(A))
 		return FALSE
 
+	if(!canPlace(A))
+		var/obj/blueprint_type = blueprint
+		to_chat(user, span_notice("There is something blocking you from placing a [initial(blueprint_type.name)] there."))
+		return
 	if(checkResource(machinery_data["cost"][blueprint], user) && blueprint)
 		//"cost" is relative to delay at a rate of 10 matter/second  (1matter/decisecond) rather than playing with 2 different variables since everyone set it to this rate anyways.
 		if(do_after(user, machinery_data["cost"][blueprint], target = A))
 			if(checkResource(machinery_data["cost"][blueprint], user) && canPlace(A))
 				useResource(machinery_data["cost"][blueprint], user)
 				activate()
 				playsound(src.loc, 'sound/machines/click.ogg', 50, TRUE)
-				new blueprint (A, FALSE, layers[current_layer])
+				if(ispath(blueprint, /obj/machinery/duct))
+					var/is_omni = current_color == DUCT_COLOR_OMNI
+					new blueprint(A, FALSE, GLOB.pipe_paint_colors[current_color], GLOB.plumbing_layers[current_layer], null, is_omni)
+				else
+					new blueprint(A, FALSE, GLOB.plumbing_layers[current_layer])
 				return TRUE
 
 /obj/item/construction/plumbing/proc/canPlace(turf/T)
```

Comment:
T ends up being the most okay single letter var name but still, better var name since you use it and all that (can even just be turf i'd be ok with that but there are probably better ideas)